### PR TITLE
feat(issue): paginate issue comments and events

### PR DESCRIPTION
**Pull request by chinesedfan**
| Question         | Response    |
| ---------------- | ----------- |
| Version?         | v1.6.0      |
| Devices tested?  | iPhone 6 Plus Simulator |
| Bug fix?         | no      |
| New feature?     | yes      |
| Includes tests?  | no      |
| All Tests pass?  | yes      |
| Related ticket?  | #751        |

---

## Screenshots

There are several screens that have been refactored.

- [x] edit-issue-comment
- [x] issue-settings
- [x] issue
- [x] new-issue
- [x] pull-merge

| Before   | After    |
| -------- | -------- |
|edit-issue-comment-old|edit-issue-comment|
|issue-old|issue|
|issue-settings-old|issue-settings|
|new-issue-old|new-issue|
|pull-merge-old|pull-merge|

## Description

This PR loads paginated issue comments and events together by the new API client. Feeling hard to keep sorting by time with two pagination APIs, so I choose the issue timeline API, even though it is still in preview version.

To make this PR as small as possible, feature improvements can include but do not limit to,

- fetching pull request information by GraphQL
- fetching repository labels by the new API client
- designing and controlling navigation params better

### CODE REVIEW

**Comment by housseindjirdeh** [comment]
This is so so great, thank you so much. Something we've needed in the app for some time for these screens :)

Will take a look by tonight 🙌

**Comment by lex111** [comment]
@chinesedfan thank you for the excellent work, but I do not yet have the opportunity to check it.

File: src/api/client.js [hunk 1]
```diff
@@ -73,6 +73,7 @@ export class Client {
     HTML: 'application/vnd.github.v3.html+json',
     JSON: 'application/vnd.github.v3+json',
     MERCY_PREVIEW: 'application/vnd.github.mercy-preview+json',
+    MOCKINGBIRD_PREVIEW: 'application/vnd.github.mockingbird-preview+json',
```
Comment (housseindjirdeh) [inline src/api/client.js]: we may want to somehow keep track of when these preview APIs end up changing

Comment (housseindjirdeh) [reply src/api/client.js]: Looks like this one opened in 2016, wow lol

https://developer.github.com/changes/2016-05-23-timeline-preview-api/

But still something to keep in mind 🤔

File: src/issue/screens/issue.screen.js [hunk 2]
```diff
@@ -314,14 +341,20 @@ class Issue extends Component {
     );
   };
 
+  renderFooter = () => {
+    return this.props.timelineItemsPagination.nextPageUrl ? (
+      <LoadingCommonItem />
```
Comment (housseindjirdeh) [inline src/issue/screens/issue.screen.js]: This is for showing the loading indicator at the bottom when user is pulling the screen up to scroll more right? Maybe we should rename this to `LoadingFooterIcon` or something similar?

Tiny nit though

Comment (chinesedfan) [reply src/issue/screens/issue.screen.js]: Naming is one of the most difficult things in computer science. 🙈 
I saw other loading components were called `loading-xx-item`, and thought this one was more in common use. BOOM, `loading-common-item`.

File: src/api/client.js (hunk 1 above)
Comment (chinesedfan) [reply src/api/client.js]: The other preview API, MERCY_PREVIEW, which was for fetching topics of repository, was replaced by GraphQL. If no so-called preview versions in GraphQL, we can also switch the timeline API.

**Pull request merged**